### ISSUE TITLE
ddb build to also pull any missing image

### ISSUE THREAD

**Issue by tonin** [perfsonar/docker-devbox#4]
It seems like `ddb build` doesn't pull images out of Docker Hub automatically, if one doesn't have an image locally.  Or is my setup missing something?

```
$ ddb build d10
[+] Building 0.6s (2/3)                                                                                                                                                                                 
[+] Building 0.6s (3/3) FINISHED                                                                                                                                                                        
 => [internal] load build definition from Dockerfile                                           0.0s
 => => transferring dockerfile: 113B                                                           0.0s
 => [internal] load .dockerignore                                                              0.0s
 => => transferring context: 2B                                                                0.0s
 => ERROR [internal] load metadata for docker.io/library/debian:10                             0.5s
------
 > [internal] load metadata for docker.io/library/debian:10:
------
failed to solve with frontend dockerfile.v0: failed to create LLB definition: rpc error:
code = Unknown desc = error getting credentials - err: exit status 1, out: ``
```

Then, doing a pull first:
```
$ docker pull debian:10
10: Pulling from library/debian
b2404786f3fe: Pull complete 
Digest: sha256:233c3bbc892229c82da7231980d50adceba4db56a08c0b7053a4852782703459
Status: Downloaded newer image for debian:10
docker.io/library/debian:10
$ ddb build d10
[+] Building 28.9s (9/9) FINISHED                                                                                                                                                                       
 => [internal] load build definition from Dockerfile                                           0.1s
 => => transferring dockerfile: 113B                                                           0.0s
 => [internal] load .dockerignore                                                              0.0s
 => => transferring context: 2B                                                                0.0s
 => [internal] load metadata for docker.io/library/debian:10                                   0.0s
 => [1/4] FROM docker.io/library/debian:10                                                     0.1s
 => [internal] load build context                                                              0.1s
 => => transferring context: 7.00kB                                                            0.0s
 => [2/4] RUN mkdir -p "/.ddb"                                                                 1.5s
 => [3/4] COPY prep /.ddb                                                                      0.1s
 => [4/4] RUN "/.ddb/prep"                                                                    26.4s
 => exporting to image                                                                         0.5s
 => => exporting layers                                                                        0.5s
 => => writing image sha256:f99a5bb0803420ba4e5b9cf7c2f1412a7423a2d802ae3cc674b0713de109e2b2   0.0s 
 => => naming to docker.io/library/ddb__antoine__d10

**Comment by mfeit-internet2** [comment]
Followup from Slack:

All `build` does is
```
$DOCKER build -t "${CONTAINER_NAME}" -f "${DOCKERFILE}" .
```
Where `$DOCKER` is either `docker` or `sudo docker`.

Something weird on your system, perhaps?

**Comment by tonin** [comment]
I'm running Docker Desktop 4.16.2 on MacOS 11.7.4 and the issue is with using sudo.  Docker on MacOS is run as a normal user.  Changing   Shouldn't it be preferable to run it as a normal user unless in some specific use case?

**Comment by tonin** [comment]
Mmmmm, not sure if it's related to running as a normal user or to the way Docker works on MacOS, but the `ddb halt` command also fails:
```
root@maddash:~# poweroff -f
Powering off.
Failed to reboot: Operation not permitted
```
or
```
root@maddash:~# poweroff --halt
System has not been booted with systemd as init system (PID 1). Can't operate.
Failed to connect to bus: Host is down
Failed to talk to init daemon.
```

**Comment by tonin** [comment]
Solved with 1da96e9
